CriticalMass/RequestManager.swift

```diff
@@ -7,51 +7,78 @@
 
 import os.log
 import UIKit
+import Network
 
 public class RequestManager {
-    private struct SendLocationPostBody: Codable {
-        var device: String
-        var location: Location
-    }
-
     private struct SendMessagePostBody: Codable {
         var device: String
         var messages: [SendChatMessage]
     }
 
     private let endpoint: URL
 
-    private var hasActiveRequest = false
-
     private var dataStore: DataStore
     private var locationProvider: LocationProvider
     private var networkLayer: NetworkLayer
     private var idProvider: IDProvider
 
+    private var pathMonitorRef: AnyObject?
+
+    private let operationQueue: OperationQueue = {
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 1
+
+        return queue
+    }()
+
     private var log = OSLog(subsystem: Bundle.main.bundleIdentifier!, category: "RequestManager")
 
-    init(dataStore: DataStore, locationProvider: LocationProvider, networkLayer: NetworkLayer, interval: TimeInterval = 12.0, idProvider: IDProvider, url: URL) {
+    @available(iOS 12.0, *)
+    convenience init(dataStore: DataStore, locationProvider: LocationProvider, networkLayer: NetworkLayer, interval: TimeInterval = 6.0, idProvider: IDProvider, url: URL, pathMonitor: NWPathMonitor) {
```

Comment:
As you sugested, I would love to see a protocol for NWPathMonitor, this would make testing easier and also remove the second initializer

Reply:
In progress